examples/cpp/hello_world/HelloWorldSubscriberWaitset.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file HelloWorldSubscriberWaitset.h
+ *
+ */
+
+#ifndef HELLO_WORLD_SUBSCRIBER_WAITSET_H_
+#define HELLO_WORLD_SUBSCRIBER_WAITSET_H_
+
+#include <fastdds/dds/core/condition/GuardCondition.hpp>
+#include <fastdds/dds/core/condition/WaitSet.hpp>
+#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/subscriber/DataReaderListener.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+
+#include "HelloWorldPubSubTypes.h"
+
+using namespace eprosima::fastdds::dds;
+
+class HelloWorldSubscriberWaitset
+{
+public:
+
+    HelloWorldSubscriberWaitset();
+
+    virtual ~HelloWorldSubscriberWaitset();
+
+    //! Run subscriber
+    void run();
+
+    //! Return the current state of execution
+    static bool is_stopped();
```

Comment:
Again, private; no statics if possible

Reply:
I've got ride of the many static as possible, but as long as calling them from threads, most of them are required.